zulipterminal/ui_tools/buttons.py

```diff
@@ -133,13 +134,17 @@ def __init__(self, controller: Any, width: int) -> None:
 
 
 class StreamButton(TopButton):
-    def __init__(self, properties: List[Any],
+    def __init__(self, properties: StreamData,
                  controller: Any, view: Any, width: int,
                  count: int=0) -> None:
         # FIXME Is having self.stream_id the best way to do this?
         # (self.stream_id is used elsewhere)
-        (self.stream_name, self.stream_id, self.color, is_private,
-         self.description) = properties
+        self.stream_name = properties['name']
+        self.stream_id = properties['id']
+        self.color = properties['color']
+        is_private = properties['invite_only']
+        self.description = properties['description']
+
```

Comment:
Possible refactor - we could just save self.properties and not have so many variables.